src/modules/barrier/barrier.c

```diff
@@ -221,14 +222,18 @@ static void enter_request_cb (flux_t *h, flux_msg_handler_t *mh,
     if (b->count == b->nprocs) {
         if (exit_event_send (ctx->h, b->name, 0) < 0)
             flux_log_error (ctx->h, "exit_event_send");
-    } else if (ctx->rank > 0 && !ctx->timer_armed) {
-        flux_timer_watcher_reset (ctx->timer, barrier_reduction_timeout_sec, 0.);
+    }
+    else if (ctx->rank > 0 && !ctx->timer_armed) {
+        flux_timer_watcher_reset (ctx->timer,
+                                  barrier_reduction_timeout_sec,
+                                  0.);
         flux_watcher_start (ctx->timer);
         ctx->timer_armed = true;
     }
-done:
-    if (sender)
-        free (sender);
+    free (sender);
+    return;
+error:
+    flux_respond_error (ctx->h, msg, errno, NULL);
 }
```

Comment:
need to `free (sender)` here too.  (Later noticed you have added it in a later commit, but should really be in commit b2c4613c7402350f9baf96b9668471df56de65d4 instead of a8103f82c4e9c9aed6315211d358397af19cb900)